src/pkg/utils/exec/exec.go

```diff
@@ -165,39 +165,39 @@ func GetOSShell(shellPref types.ZarfComponentActionShell) (string, string) {
 			shell = shellPref.Windows
 		}
 
-		shellArgs = "-Command"
+		shellArgs = append(shellArgs, "-Command", "$ErrorActionPreference = 'Stop';")
 		if shell == "cmd" {
 			// Change shellArgs to /c if cmd is chosen
-			shellArgs = "/c"
+			shellArgs = append(shellArgs, "/c")
 		} else if !IsPowershell(shell) {
 			// Change shellArgs to -c if a real shell is chosen
-			shellArgs = "-c"
+			shellArgs = append(shellArgs, "-e", "-c")
 		}
 	case "darwin":
 		shell = "sh"
 		if shellPref.Darwin != "" {
 			shell = shellPref.Darwin
 		}
 
-		shellArgs = "-c"
+		shellArgs = append(shellArgs, "-e", "-c")
```

Comment:
```suggestion
		shellArgs = "-ce")
```
Shell args can be combined on POSIX (real) shells and then the array complexity might not be needed.